src/Type/ObjectProphecy/WillExtendOrImplementDynamicReturnTypeExtension.php

```diff
@@ -87,14 +95,16 @@ public function getTypeFromMethodCall(
             $className = $scope->getClassReflection()->getName();
         }
 
+        \assert(\get_class($calledOnType) === Type\Generic\GenericObjectType::class);
```

Comment:
This is to avoid `assert($calledOnType instanceof Type\Generic\GenericObjectType::class);`
( see https://phpstan.org/blog/why-is-instanceof-type-wrong-and-getting-deprecated )
.. I'm not sure this assertion check is good, or not.

Reply:
Yes, that's absolutely correct. I think I've been relying too much on workarounds.

Looking at PHPStan's own baseline, I noticed that a lot of phpstanApi.instanceofType remains. I think I'll revert it back to instanceof.
https://github.com/phpstan/phpstan-src/blob/2.0.2/phpstan-baseline.neon